sdk/monitor/Azure.Monitor.Query/src/LogsClient.cs

```diff
@@ -87,15 +89,80 @@ public virtual LogsBatchQuery CreateBatchQuery()
             return new LogsBatchQuery(_clientDiagnostics, _queryClient, _rowBinder);
         }
 
-        internal static QueryBody CreateQueryBody(string query, TimeSpan? timeSpan)
+        internal static QueryBody CreateQueryBody(string query, TimeSpan? timeSpan, LogsQueryOptions options, out string prefer)
         {
             var queryBody = new QueryBody(query);
             if (timeSpan != null)
             {
                 queryBody.Timespan = TypeFormatters.ToString(timeSpan.Value, "P");
             }
 
+            StringBuilder preferBuilder = null;
+            if (options?.Timeout is TimeSpan timeout)
+            {
+                preferBuilder ??= new();
+                preferBuilder.Append("wait=");
+                preferBuilder.Append((int) timeout.TotalSeconds);
+            }
+
+            if (options?.IncludeStatistics == true)
+            {
+                preferBuilder ??= new();
+                preferBuilder.Append(" include-statistics=true");
+            }
+
+            prefer = preferBuilder?.ToString();
+
             return queryBody;
         }
+
+        private async Task<Response<LogsQueryResult>> ExecuteAsync(string workspaceId, string query, TimeSpan? timeSpan, LogsQueryOptions options, bool async, CancellationToken cancellationToken = default)
+        {
+            if (workspaceId == null)
+            {
+                throw new ArgumentNullException(nameof(workspaceId));
+            }
+
+            QueryBody queryBody = CreateQueryBody(query, timeSpan, options, out string prefer);
+            using var message = _queryClient.CreateExecuteRequest(workspaceId, queryBody, prefer);
+
+            if (options?.Timeout != null)
+            {
+                message.SetProperty("NetworkTimeoutOverride", options.Timeout);
```

Comment:
Should we make "NetworkTimeoutOverride" a known const somewhere?

Reply:
I'll turn it into an actual property.